components/sup/src/ctl_gateway/mod.rs

```diff
@@ -79,6 +73,7 @@ pub struct CtlRequest {
     /// Transaction for the given request.
     transaction: Option<protocol::codec::SrvTxn>,
     current_color_spec: Option<ColorSpec>,
+    is_new_line: bool,
```

Comment:
As long as we're changing this struct, I think we could eliminate the `Option` around `current_color_spec` and replace the `None` cases with `ColorSpec::default`. That should simplify a bunch of the logic, I think.